src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlNetworkTranslator.cs

```diff
@@ -46,11 +46,21 @@ public class NpgsqlNetworkTranslator : IMethodCallTranslator
         [CanBeNull]
         public Expression Translate(MethodCallExpression expression)
         {
-            if (expression.Method.DeclaringType != typeof(NpgsqlNetworkExtensions))
+            var declaringType = expression.Method.DeclaringType;
```

Comment:
I'd prefer it if we handled the `IPAddress` and `PhysicalAddress` methods separately, rather than include them in the big switch below. As it is, in theory, if a method called `LessThan` (or any other method below) existed on `IPAddress` or `PhysicalAddress`, it would get mistakenly translated. So I'd first take care of the `Parse()` cases, and then move ahead as before to the switch over all `NpgsqlNetworkExtensions` methods.

I do admit this is probably a bit of theoretical OCD on my side :)

Reply:
I like that idea. I'd prefer to still use one translator class though, but with the `Parse(string)` handling separated from the main `switch`.

Reply:
Absolutely, no need for multiple translators.